crates/gitbutler-branch-actions/src/base.rs

```diff
@@ -604,10 +647,45 @@ pub(crate) fn target_to_base_branch(ctx: &CommandContext, target: &Target) -> Re
             .map(FetchResult::timestamp)
             .copied()
             .map(|t| t.duration_since(time::UNIX_EPOCH).unwrap().as_millis()),
+        diverged,
+        diverged_fork_point,
+        diverged_ahead,
+        diverged_behind,
     };
     Ok(base)
 }
 
 fn default_target(base_path: &Path) -> Result<Target> {
     VirtualBranchesHandle::new(base_path).get_default_target()
 }
+
+pub(crate) fn push(ctx: &CommandContext, with_force: bool) -> Result<()> {
+    ctx.assure_resolved()?;
+    let target = default_target(&ctx.project().gb_dir())?;
+    let _ = ctx.push(target.sha, &target.branch, with_force, None, None);
+    Ok(())
+}
+
+pub(crate) fn realign_base_branch(
+    ctx: &CommandContext,
+    perm: &mut WorktreeWritePermission,
+) -> anyhow::Result<Vec<ReferenceName>> {
+    ctx.assure_resolved()?;
+
+    let target = default_target(&ctx.project().gb_dir())?;
+    let base = target_to_base_branch(ctx, &target)?;
+
+    if !base.diverged {
+        return Ok(Vec::new());
+    }
+
+    let target_commit = ctx
+        .repository()
+        .find_commit(
+            base.diverged_fork_point
+                .context("failed to find fork point")?,
+        )
+        .context("failed to find target commit")?;
+
+    set_base_branch_to_commit(ctx, target_commit, perm)
+}
```

Comment:
What is the difference between update_base_branch and realign_base_branch? Can they be made the same function?

Reply:
`update_base_branch`: 
This resets the local base branch to whatever the tip of its upstream is.

`realign_base_branch`:
This resets the local base branch to the fork-point it has to its upstream.
This drops any ahead changes in the local base branch.
So it's "aligned" in the sense that it no longer diverts from upstream.